src/librustc_typeck/coherence/mod.rs

```diff
@@ -250,29 +250,44 @@ impl<'a, 'gcx, 'tcx> CoherenceChecker<'a, 'gcx, 'tcx> {
 
             match param_env.can_type_implement_copy(tcx, self_type, span) {
                 Ok(()) => {}
-                Err(CopyImplementationError::InfrigingField(name)) => {
-                    struct_span_err!(tcx.sess,
-                                     span,
-                                     E0204,
-                                     "the trait `Copy` may not be implemented for this type")
-                        .span_label(span, &format!("field `{}` does not implement `Copy`", name))
-                        .emit()
+                Err(CopyImplementationError::InfrigingField(did, name)) => {
+                    let mut err = struct_span_err!(tcx.sess,
+                                                   span,
+                                                   E0204,
+                                                   "the trait `Copy` may not be implemented for \
+                                                   type `{}`",
+                                                   self_type);
+                    err.span_label(span,
+                                   &format!("field `{}` in struct `{}` doesn't implement `Copy`",
```

Comment:
Maybe we can say the "in struct {}" part if we don't have the span on line 264 available, otherwise we can remove it because it it's redundant with that label